Makefile

```diff
@@ -120,181 +134,141 @@ build/src/libultra/libc/ll%: OPTFLAGS := -O1
+	$(ELF2ROM) -cic 6105 $< $@
 
-asm/%.asm: asm/disasm.dep ;
+$(ROMC): $(ROM)
+	python3 tools/z64compress_wrapper.py --mb 32 --matching --threads $(shell nproc) $< $@ $(ELF) build/$(SPEC)
 
-asm/disasm.dep: tables/files.txt tables/functions.txt tables/objects.txt tables/variables.txt tables/vrom_variables.txt
-	@mkdir -p asm
-	@touch $@
-	./tools/disasm.py -d ./asm -l ./tables/files.txt -f ./tables/functions.txt -o ./tables/objects.txt -v ./tables/variables.txt -v ./tables/vrom_variables.txt || rm $@
+$(ELF): $(TEXTURE_FILES_OUT) $(OVERLAY_RELOC_FILES) $(O_FILES) build/ldscript.txt build/undefined_syms.txt
+	$(LD) -T build/undefined_syms.txt -T build/ldscript.txt --no-check-sections --accept-unknown-input-arch --emit-relocs -Map build/mm.map -o $@
 
 
 #### Main commands ####
 
 ## Cleaning ##
 clean:
-	$(RM) -rf $(ROM) $(UNCOMPRESSED_ROM) build
+	$(RM) -rf $(ROMC) $(ROM) $(ELF) build
 
 assetclean:
 	$(RM) -rf $(ASSET_BIN_DIRS)
 	$(RM) -rf build/assets
+	$(RM) -rf .extracted-assets.json
 
 distclean: assetclean clean
-	$(RM) -rf baserom/ asm/ expected/
+	$(RM) -rf asm baserom data
 	$(MAKE) -C tools clean
 
 ## Extraction step
 setup:
 	$(MAKE) -C tools
-	./tools/extract_rom.py $(MM_BASEROM)
+	python3 tools/fixbaserom.py
+	python3 tools/extract_baserom.py
 	python3 extract_assets.py
 
 ## Assembly generation
-assembly: $(S_FILES)
-	@echo "Assembly generated."
+disasm:
+	$(RM) -rf asm data
+	python3 tools/disasm/disasm.py
 
 diff-init: all
 	$(RM) -rf expected/
 	mkdir -p expected/
 	cp -r build expected/build
-	cp $(UNCOMPRESSED_ROM) expected/$(UNCOMPRESSED_ROM)
-	cp $(ROM) expected/$(ROM)
 
 init:
 	$(MAKE) distclean
 	$(MAKE) setup
-	$(MAKE) assembly
+	$(MAKE) disasm
 	$(MAKE) all
 	$(MAKE) diff-init
 
-# Recipes
-
-build/baserom:
-	cp -r baserom/ build/baserom/
-
-# FIXME: The process of splitting rodata changes the assembly files, so we must avoid making .o files for them until that is done.
-# The simplest way to do that is to give them an order dependency on .c files' .o files
-build/asm/%.o: asm/%.asm | $(C_O_FILES)
-	iconv --from UTF-8 --to EUC-JP $^ | $(AS) $(ASFLAGS) -o $@
-
-build/src/overlays/%.o: src/overlays/%.c
-	$(CC) -c $(CFLAGS) $(MIPS_VERSION) $(OPTFLAGS) -o $@ $<
-	./tools/overlay.py $@ build/src/overlays/$*_overlay.s
-	$(AS) $(ASFLAGS) build/src/overlays/$*_overlay.s -o build/src/overlays/$*_overlay.o
-
-build/%.o: %.c
-	$(CC) -c $(CFLAGS) $(MIPS_VERSION) $(OPTFLAGS) -o $@ $<
-
 build/assets/%.o: assets/%.c
 	$(CC) -I build/ -c $(CFLAGS) $(MIPS_VERSION) $(OPTFLAGS) -o $@ $<
 
-build/src/libultra/libc/ll.o: src/libultra/libc/ll.c
-	$(CC) -c $(CFLAGS) $(MIPS_VERSION) $(OPTFLAGS) -o $@ $<
-	@./tools/set_o32abi_bit.py $@
-
-build/src/libultra/libc/llcvt.o: src/libultra/libc/llcvt.c
-	$(CC) -c $(CFLAGS) $(MIPS_VERSION) $(OPTFLAGS) -o $@ $<
-	@./tools/set_o32abi_bit.py $@
-
-build/comp/%.yaz0: build/binary/%
-	./tools/yaz0 $< $@
+build/binary/assets/scenes/%: build/code.elf
+	@$(OBJCOPY) --dump-section $*=$@ $< /dev/null
```

Comment:
What is this doing?

Reply:
Nothing, it seems: `build/code.elf` is never made, so the rule never runs. The `binary` directory is never used any more, so this can just be removed?